plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.test.ts

```diff
@@ -73,6 +77,153 @@ describe('router', () => {
     });
   });
 
+  describe('find application', () => {
+    it(`successfully finds application based on name`, async () => {
+      getArgoAppData.mockResolvedValue({
+        items: [
+          {
+            metadata: {
+              name: 'test-app-prod',
+              namespace: 'argocd',
+              labels: {
+                app: 'test-app',
+              },
+            },
+          },
+          {
+            metadata: {
+              name: 'test-app-staging',
+              namespace: 'argocd',
+              labels: {
+                app: 'test-app',
+              },
+            },
+          },
+        ],
+      });
+
+      const response = await request(app).get('/find/name/test-app');
+
+      expect(getArgoToken).toBeCalledTimes(2);
+      expect(response.body).toMatchObject([
+        {
+          appName: ['test-app'],
+          name: 'argoInstance1',
+          url: 'https://argoInstance1.com',
+        },
+        {
+          appName: ['test-app'],
+          name: 'argoInstance2',
+          url: 'https://argoInstance2.com',
+        },
+      ]);
+    });
+
+    it(`successfully finds no application based on name`, async () => {
+      getArgoAppData.mockRejectedValue('error getting application');
+
+      const response = await request(app).get('/find/name/test-app');
+
+      expect(getArgoToken).toBeCalledTimes(2);
+      expect(response.body).toMatchObject([]);
+    });
+
+    it(`successfully finds application based on selector`, async () => {
+      getArgoAppData.mockResolvedValue({
+        items: [
+          {
+            metadata: {
+              name: 'test-app-prod',
+              namespace: 'argocd',
+              labels: {
+                app: 'test-app',
+              },
+            },
+          },
+          {
+            metadata: {
+              name: 'test-app-staging',
+              namespace: 'argocd',
+              labels: {
+                app: 'test-app',
+              },
+            },
+          },
+        ],
+      });
+
+      const selector = encodeURIComponent(
+        'prefix/name=test-app,other-prefix/name=some-value',
+      );
+      const response = await request(app).get(`/find/selector/${selector}`);
+
+      expect(getArgoAppData).toBeCalledTimes(2);
+      expect(getArgoAppData).toBeCalledWith(
+        'https://argoInstance1.com',
+        'argoInstance1',
+        { selector: 'prefix/name=test-app,other-prefix/name=some-value' },
```

Comment:
MINOR: A suggestion to make it obvious that the `selector` value is identical.

```suggestion
      const selector = 'prefix/name=test-app,other-prefix/name=some-value';
  
      const response = await request(app).get(`/find/selector/${
        encodeURIComponent(selector as string)
      }`);

      expect(getArgoAppData).toBeCalledTimes(2);
      expect(getArgoAppData).toBeCalledWith(
        'https://argoInstance1.com',
        'argoInstance1',
        { selector },
```

Reply:
Thx. You are of course right. Fixed and pushed.